From 30a22d05dc4034515288a757c80252a7a2c8541f Mon Sep 17 00:00:00 2001
From: wangguosheng <wangguosheng@eswincomputing.com>
Date: Fri, 15 Aug 2025 18:36:01 +0800
Subject: [PATCH 466/533] fix: d560 enables 4-channel fan

Changelogs:
1. d560 test whether the function of the 4-channel fan is normal.

Signed-off-by: wangguosheng <wangguosheng@eswincomputing.com>
---
 .../dts/eswin/eic7702-d560-interleave.dts     | 25 +++++++++++++++++++
 arch/riscv/boot/dts/eswin/eic7702-d560.dts    | 21 ++++++++++++++++
 .../dts/eswin/eic7702-som560-interleave.dtsi  |  2 +-
 3 files changed, 47 insertions(+), 1 deletion(-)

diff --git a/arch/riscv/boot/dts/eswin/eic7702-d560-interleave.dts b/arch/riscv/boot/dts/eswin/eic7702-d560-interleave.dts
index 8677088b4728..663f157a57ae 100644
--- a/arch/riscv/boot/dts/eswin/eic7702-d560-interleave.dts
+++ b/arch/riscv/boot/dts/eswin/eic7702-d560-interleave.dts
@@ -174,6 +174,31 @@ aon_eeprom@50 {
 	};
 };
 
+&fan_control {
+	status = "okay";
+};
+
+&d1_fan_control {
+	pinctrl-0 = <&pinctrl_die1_fan_tach_default &pinctrl_die1_pwm0_default
+				&pinctrl_die1_pwm1_default &pinctrl_die1_pwm2_default>;
+	status = "okay";
+	fan-0 {
+		label = "d1_fan";
+		pwms = <&d1_pwm0 0 100000 0>;
+		status = "okay";
+	};
+	fan-1 {
+		label = "chassis_fan1";
+		pwms = <&d1_pwm0 1 100000 0>;
+		status = "okay";
+	};
+	fan-2 {
+		label = "chassis_fan2";
+		pwms = <&d1_pwm0 2 100000 0>;
+		status = "okay";
+	};
+};
+
 &d0_i2c1 {
 	/* pin header */
 	status = "okay";
diff --git a/arch/riscv/boot/dts/eswin/eic7702-d560.dts b/arch/riscv/boot/dts/eswin/eic7702-d560.dts
index 32af941861d6..2025ce58d4f1 100644
--- a/arch/riscv/boot/dts/eswin/eic7702-d560.dts
+++ b/arch/riscv/boot/dts/eswin/eic7702-d560.dts
@@ -178,6 +178,27 @@ &fan_control {
 	status = "okay";
 };
 
+&d1_fan_control {
+	pinctrl-0 = <&pinctrl_die1_fan_tach_default &pinctrl_die1_pwm0_default
+				&pinctrl_die1_pwm1_default &pinctrl_die1_pwm2_default>;
+	status = "okay";
+	fan-0 {
+		label = "d1_fan";
+		pwms = <&d1_pwm0 0 100000 0>;
+		status = "okay";
+	};
+	fan-1 {
+		label = "chassis_fan1";
+		pwms = <&d1_pwm0 1 100000 0>;
+		status = "okay";
+	};
+	fan-2 {
+		label = "chassis_fan2";
+		pwms = <&d1_pwm0 2 100000 0>;
+		status = "okay";
+	};
+};
+
 &d0_i2c1 {
 	/* pin header */
 	status = "okay";
diff --git a/arch/riscv/boot/dts/eswin/eic7702-som560-interleave.dtsi b/arch/riscv/boot/dts/eswin/eic7702-som560-interleave.dtsi
index 20926a20be8b..783745d195ad 100644
--- a/arch/riscv/boot/dts/eswin/eic7702-som560-interleave.dtsi
+++ b/arch/riscv/boot/dts/eswin/eic7702-som560-interleave.dtsi
@@ -1232,7 +1232,7 @@ &pwm0 {
 	fan-gpios = <&portc 4 GPIO_ACTIVE_HIGH>;
 };
 &d1_pwm0 {
-	status = "disabled";
+	status = "okay";
 };
 
 &pvt0 {
-- 
2.52.0

